### ISSUE TITLE
PWSH 7.2.1: invoke-webrequest hangs at: VERBOSE: received -byte response of content type text/plain

### ISSUE THREAD

**Issue by penalvch-zz** [PowerShell/PowerShell#16685]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

While attempting to download a file from the following URL, invoke-webrequest hangs seemingly indefinitely with no indication of why. The 'Reading Response' displays, so it seems to actually download the data, but just hangs after doing so. The PWSH window is not responsive to Ctrl+C.

*NOTE*: If testing the below, replace the useragent string with a valid 'Company Name' and 'email address' as per below U.S. sec.gov Webmaster FAQs. When I ran the command I use my own email address with my own custom string (changing the user agent string to something completely different didn't change anything):
https://www.sec.gov/os/webmaster-faq#developers

However, the following equivalent Windows PowerShell 5.1 command works perfectly so seems regression:
$test=invoke-webrequest -uri 'https://www.sec.gov/Archives/edgar/data/0000008177/000156761917000512/0001567619-17-000512.txt' -useragent 'Private Investor EMAIL@ADDRESS.COM' -usebasicparsing -method get -verbose
PS C:\WINDOWS\system32>

WORKAROUND on Windows only:
$test=& 'powershell.exe' "(invoke-webrequest -uri 'https://www.sec.gov/Archives/edgar/data/0000008177/000156761917000512/0001567619-17-000512.txt' -useragent 'Private Investor EMAIL@ADDRESS.COM' -usebasicparsing -method get).content"

### Expected behavior

```console
PS C:\Windows\System32>$test=invoke-webrequest -uri 'https://www.sec.gov/Archives/edgar/data/0000008177/000156761917000512/0001567619-17-000512.txt' -useragent 'Private Investor YOUREMAILADDRESS@EMAILDOMAIN.COM' -SkipHeaderValidation -method GET -verbose
PS C:\Windows\System32>
```

### Actual behavior

```console
PS C:\Windows\System32>$test=invoke-webrequest -uri 'https://www.sec.gov/Archives/edgar/data/0000008177/000156761917000512/0001567619-17-000512.txt' -useragent 'Private Investor YOUREMAILADDRESS@EMAILDOMAIN.COM' -SkipHeaderValidation -method GET -verbose
VERBOSE: GET with 0-byte payload
VERBOSE: received -byte response of content type text/plain
```

### Environment data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by penalvch** [comment]
Same problem with:
https://www.sec.gov/Archives/edgar/data/0000810509/000143774919022151/0001437749-19-022151.txt

**Comment by iSazonov** [comment]
@TravisEz13 @PaulHigin Maybe for WG-Security. Infinite cycle in Regex https://github.com/PowerShell/PowerShell/blob/a5d41c22871c4835de2a0f8ba17d553fa777c6a3/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs#L435

**Comment by PaulHigin** [comment]
@WG-Security
WG-Security group looked at this and there is potential for infinite recursion on the regex (s_metaexp) string (line 414) and DOS attack.
There should be fuzz testing along with a fix, and this can be done by using text files for fuzzing source.

Note that dotNet has added code to remove some potential infinite recursions, when possible.  But we shouldn't count on it.

**Comment by CarloToso** [comment]
@iSazonov This issue is fixed in `PowerShell 7.4.0 preview 1`

**Comment by iSazonov** [comment]
Fixed in #17813
